integration_tests/utils.py

```diff
@@ -261,11 +284,181 @@ def revoke_fee_grant(cli, granter_address, grantee, i=0, *k_options, **kv_option
+            keyring_backend="test",
+            chain_id=cli.cosmos_cli(i).chain_id,
+            node=cli.cosmos_cli(i).node_rpc,
             **kv_options,
         )
     )
+
+
+@throw_error_for_non_success_code
+def revoke_authorization(
+    cli, grantee, msg_type, granter, *k_options, i=0, **kv_options
+):
+    return json.loads(
+        cli.cosmos_cli(i).raw(
+            "tx",
+            "authz",
+            "revoke",
+            grantee,
+            msg_type,
+            "-y",
+            *k_options,
+            from_=granter,
+            home=cli.cosmos_cli(i).data_dir,
+            keyring_backend="test",
+            chain_id=cli.cosmos_cli(i).chain_id,
+            node=cli.cosmos_cli(i).node_rpc,
+            **kv_options,
+        )
+    )
+
+
+def query_command(cli, *k_options, i=0, **kv_options):
+    return json.loads(
+        cli.cosmos_cli(i).raw(
+            "query",
+            *k_options,
+            home=cli.cosmos_cli(i).data_dir,
+            chain_id=cli.cosmos_cli(i).chain_id,
+            node=cli.cosmos_cli(i).node_rpc,
+            output="json",
+            *kv_options,
+        )
+    )
+
+
+@throw_error_for_non_success_code
+def delegate_amount(
+    cli, validator_address, amount, from_, *k_options, i=0, **kv_options
+):
+    return json.loads(
+        cli.cosmos_cli(i).raw(
+            "tx",
+            "staking",
+            "delegate",
+            validator_address,
+            amount,
+            "-y",
+            *k_options,
+            from_=from_,
+            home=cli.cosmos_cli(i).data_dir,
+            keyring_backend="test",
+            chain_id=cli.cosmos_cli(i).chain_id,
+            node=cli.cosmos_cli(i).node_rpc,
+            **kv_options,
+        )
+    )
+
+
+@throw_error_for_non_success_code
+def unbond_amount(cli, validator_address, amount, from_, *k_options, i=0, **kv_options):
+    return json.loads(
+        cli.cosmos_cli(i).raw(
+            "tx",
+            "staking",
+            "unbond",
+            validator_address,
+            amount,
+            "-y",
+            *k_options,
+            from_=from_,
+            home=cli.cosmos_cli(i).data_dir,
+            keyring_backend="test",
+            chain_id=cli.cosmos_cli(i).chain_id,
+            node=cli.cosmos_cli(i).node_rpc,
+            **kv_options,
+        )
+    )
+
+
+@throw_error_for_non_success_code
+def redelegate_amount(
+    cli, src_validator, dst_validator, amount, from_, *k_options, i=0, **kv_options
+):
+    return json.loads(
+        cli.cosmos_cli(i).raw(
+            "tx",
+            "staking",
+            "redelegate",
+            src_validator,
+            dst_validator,
+            amount,
+            "-y",
+            *k_options,
+            from_=from_,
+            home=cli.cosmos_cli(i).data_dir,
+            keyring_backend="test",
+            chain_id=cli.cosmos_cli(i).chain_id,
+            node=cli.cosmos_cli(i).node_rpc,
```

Comment:
```suggestion
```

I think we can neglect these parameters in cli in cosmos-sdk 0.43, because they are already set in the `client.toml`, set `home` alone will include all those options.

Reply:
Yes, tried locally and verified that they can be removed. I will make the change.
Previously, I copied this code chunk from pystarport. Thank you for the advice 👍